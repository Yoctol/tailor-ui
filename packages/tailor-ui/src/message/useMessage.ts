import { useContext } from 'react';

import EffectMessageContext from './EffectMessageContext';

const useMessage = () => {
  const triggerRef = useContext(EffectMessageContext);

  return {
    info: (content: string, duration = 3000) =>
      triggerRef.current({ content, duration }, 'info'),
    success: (content: string, duration = 3000) =>
      triggerRef.current({ content, duration }, 'success'),
    warning: (content: string, duration = 3000) =>
      triggerRef.current({ content, duration }, 'warning'),
    error: (content: string, duration = 3000) =>
      triggerRef.current({ content, duration }, 'error'),
  };
};

export { useMessage };
