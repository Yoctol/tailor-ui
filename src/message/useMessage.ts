import { ReactNode, useContext } from 'react';

import { HooksMessageContext } from './HooksMessageProvider';

const useMessage = () => {
  const triggerRef = useContext(HooksMessageContext);

  return {
    info: (content: ReactNode, duration = 3000) =>
      triggerRef.current({ content, duration }, 'info'),
    success: (content: ReactNode, duration = 3000) =>
      triggerRef.current({ content, duration }, 'success'),
    warning: (content: ReactNode, duration = 3000) =>
      triggerRef.current({ content, duration }, 'warning'),
    error: (content: ReactNode, duration = 3000) =>
      triggerRef.current({ content, duration }, 'error'),
  };
};

export { useMessage };
