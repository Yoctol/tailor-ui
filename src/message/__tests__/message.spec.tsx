import React from 'react';
import userEvent from '@testing-library/user-event';

import { render, screen, waitForElementToBeRemoved } from 'test/test-utils';

import { useMessage } from '../useMessage';

const Message = () => {
  const message = useMessage();

  return (
    <button type="button" onClick={() => message.success('Success', 1000000)}>
      button
    </button>
  );
};

describe('useMessage', () => {
  it('should render message correctly', async () => {
    const { baseElement } = render(<Message />);

    const button = screen.getByText('button');
    userEvent.click(button);

    await screen.findByText('Success');

    expect(baseElement).toMatchSnapshot();
  });

  // FIXME:
  it.skip('should remove message when click close button', async () => {
    const { baseElement } = render(<Message />);

    const button = screen.getByText('button');
    userEvent.click(button);

    await screen.findByText('Success');

    const closeButton = baseElement.querySelector('i[role=button]');
    userEvent.click(closeButton as Element);

    await waitForElementToBeRemoved(() => screen.queryByText('Success'));
  });
});
