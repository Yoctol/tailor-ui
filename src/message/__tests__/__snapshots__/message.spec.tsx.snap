// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useMessage should render message correctly 1`] = `
.c4 {
  box-sizing: border-box;
  -webkit-flex: auto;
  -ms-flex: auto;
  flex: auto;
}

.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  line-height: 1;
  cursor: default;
  margin-right: 0.5rem;
}

.c3 svg {
  fill: #8b8b8b;
  vertical-align: middle;
  pointer-events: none;
  -webkit-transition: all 200ms ease 0s;
  transition: all 200ms ease 0s;
}

.c3.c3.c3 svg {
  fill: #40d064;
  width: 20px;
  height: 20px;
}

.c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  line-height: 1;
  cursor: pointer;
}

.c5 svg {
  fill: #8b8b8b;
  vertical-align: middle;
  pointer-events: none;
  -webkit-transition: all 200ms ease 0s;
  transition: all 200ms ease 0s;
}

.c5.c5.c5 svg {
  fill: #ffffff;
  width: 16px;
  height: 16px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  right: 10px;
  bottom: 10px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  position: relative;
  box-sizing: border-box;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  width: 280px;
  overflow: hidden;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  margin-top: 0.5rem;
  padding: 1rem;
  overflow: hidden;
  border-radius: 0.5rem;
  background-color: #1a1a1a;
  color: #ffffff;
  font-size: 0.875rem;
}

.c6 {
  position: absolute;
  z-index: 1;
  bottom: 0;
  left: 0;
  width: auto;
  height: 3px;
  background-color: #6B88c5;
}

<body>
  <div>
    <button
      data-testid="button"
      type="button"
    >
      button
    </button>
  </div>
  <div
    portal-container=""
  >
    <div
      style="position: fixed; top: 0px; left: 0px; z-index: 30;"
    >
      <div
        class="c0"
      >
        <div
          class="c1"
          style="opacity: 0; height: 0px;"
        >
          <div
            class="c2"
          >
            <i
              class="c3"
              cursor="default"
              fill="success"
              size="20"
            >
              <svg
                height="32"
                viewBox="0 0 32 32"
                width="32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16 31C7.716 31 1 24.284 1 16 1 7.716 7.716 1 16 1c8.284 0 15 6.716 15 15 0 8.284-6.716 15-15 15zm6.113-20.19l-9.066 8.73-3.017-3.586a2 2 0 0 0-3.06 2.576l3.705 4.403a3 3 0 0 0 4.227.364c.076-.066.076-.066.15-.135l9.835-9.471a2 2 0 1 0-2.774-2.882z"
                  fill-rule="nonzero"
                />
              </svg>
            </i>
            <div
              class="c4"
            >
              Success
            </div>
            <i
              class="c5"
              cursor="pointer"
              fill="light"
              role="button"
              size="16"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </i>
            <div
              class="c6"
              style="right: 100%;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
